redditdownload/redditdownload.py

```diff
@@ -298,21 +318,72 @@ def slugify(value):
     # with some modification
     import unicodedata
     value = unicodedata.normalize('NFKD', value).encode('ascii', 'ignore')
-    value = unicode(re.sub(r'[^\w\s-]', '', value).strip())
+    value = str(re.sub(r'[^\w\s-]', '', value.decode('ascii')).strip())
     # value = re.sub(r'[-\s]+', '-', value) # not replacing space with hypen
     return value
 
 
+def remove_extension(mystr):
+    """ Returns filename found in mystr by locating image file extension """
+    exts = ['.png', '.jpg', 'webm', '.jpeg', '.jfif', '.gif', 'gifv', '.bmp',
+            '.tif', '.tiff', '.webp', '.bpg', '.bat',
+            '.heif', '.exif', '.ppm', '.cgm', '.svg']
+    for e in exts:
+        ext_index = mystr.find(e)
+        if ext_index != -1:
+            return mystr[:ext_index]
+    return mystr
+
+
+def history_log(wdir, file_log, mode='read', write_data={}):
+    '''
```

Comment:
yes, and put it in doc. other option is to make your own Error class and make this function raise it when something wrong.